### PR TITLE
feat(GroupTheory/GroupAction/Blocks): order iso between blocks and subgroups

### PR DESCRIPTION
**Pull request by AntoineChambert-Loir**
This PR proves the order isomorphism between blocks containing a point and subgroups containing its stabilizer.

---

- [x] depends on: #12023 
- [x] depends on: #12049
[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/GroupTheory/GroupAction/Blocks.lean [hunk 1]
```diff
@@ -382,6 +382,114 @@ theorem IsBlockSystem.of_normal {N : Subgroup G} [N.Normal] :
...
+    apply smul_orbit_subset
+  simp only [Submonoid.smul_def, Subgroup.coe_toSubmonoid,
+    Set.mem_smul_set_iff_inv_smul_mem, ← mul_smul] at hb'
+  obtain ⟨⟨k, k_mem⟩, hk⟩ := hb'
+  simp only [Submonoid.mk_smul] at hk
+  rw [MulAction.mul_smul, ← smul_eq_iff_eq_inv_smul, ← inv_inv h,
+    ← smul_eq_iff_eq_inv_smul, ← mul_smul, ← mul_smul, ← mem_stabilizer_iff] at hk
+  apply hH at hk
+  rwa [Subgroup.mul_mem_cancel_right H k_mem,
+    Subgroup.mul_mem_cancel_left H (H.inv_mem h_mem)] at hk
```
Comment (tb65536) [inline Mathlib/GroupTheory/GroupAction/Blocks.lean:411]: Here's a slightly shorter proof.
```suggestion
  simp_rw [IsBlock.def_one, or_iff_not_imp_right, Set.not_disjoint_iff]
  rintro g ⟨-, ⟨-, ⟨h₁, rfl⟩, h⟩, ⟨h₂, rfl⟩⟩
  simp_rw [← inv_smul_eq_iff, H.smul_def, ← mul_smul] at h
  specialize hH h
  rw [mul_mem_cancel_left h₂⁻¹.2, mul_mem_cancel_right h₁.2] at hH
  rw [← Subgroup.coe_mk H g hH, ← H.smul_def, smul_orbit (⟨g, hH⟩ : H) a]
```

File: Mathlib/GroupTheory/GroupAction/Blocks.lean [hunk 2]
```diff
@@ -382,6 +382,114 @@ theorem IsBlockSystem.of_normal {N : Subgroup G} [N.Normal] :
...
+  · intro hg
+    suffices g • a ∈ orbit H a by
+      rw [mem_orbit_iff] at this
+      obtain ⟨k, hk⟩ := this
+      rw [smul_eq_iff_eq_inv_smul] at hk
+      rw [← Subgroup.mul_mem_cancel_left H (SetLike.coe_mem k⁻¹)]
+      apply hH
+      rw [mem_stabilizer_iff, MulAction.mul_smul, ← Submonoid.smul_def, ← hk]
+    rw [← hg]
+    simp only [Set.smul_mem_smul_set_iff, mem_orbit_self]
```
Comment (tb65536) [inline Mathlib/GroupTheory/GroupAction/Blocks.lean:453]:
```suggestion
    obtain ⟨-, ⟨b, rfl⟩, h⟩ := hg.symm ▸ mem_orbit_self a
    simp_rw [H.smul_def, ← mul_smul, ← mem_stabilizer_iff] at h
    exact (mul_mem_cancel_right b.2).mp (hH h)
```

File: Mathlib/GroupTheory/GroupAction/Blocks.lean [hunk 3]
```diff
@@ -382,6 +379,97 @@ theorem IsBlockSystem.of_normal {N : Subgroup G} [N.Normal] :
...
+    exact (mul_mem_cancel_right b.2).mp (hH h)
+  · intro hg
+    rw [mem_stabilizer_iff, ← Subgroup.coe_mk H g hg, ← Submonoid.smul_def]
+    apply smul_orbit
+
+variable (G)
+
+example (B : Set X) (k : G) (hk : k ∈ stabilizer G B) (a : X) (ha : a ∈ B) :
+  k • a ∈ B  :=
+  hk.symm ▸ Set.smul_mem_smul_set ha
```
Comment (tb65536) [inline Mathlib/GroupTheory/GroupAction/Blocks.lean:445]:
```suggestion
```

**Comment by AntoineChambert-Loir** [comment]
something went wrong with the merge, there shouldn't be thousands of modified files…

**Comment by AntoineChambert-Loir** [comment]
As I can't correct this merge, I opened a new branch #13995

**Pull request closed without merging**